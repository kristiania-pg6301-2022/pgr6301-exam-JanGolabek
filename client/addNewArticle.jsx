import React, { useState } from "react";

function FormInput({ label, value, onChangeValue }) {
    return (
        <div>
            <label>
                <strong>{label}</strong>{" "}
                <input value={value} onChange={(e) => onChangeValue(e.target.value)} />
            </label>
        </div>
    );
}

export function AddNewArticle() {
    const [category, setCategory] = useState("");
    const [title, setTitle] = useState("");
    const [author, setAuthor] = useState("");
    const [text, setText] = useState("");
    return (
        <form>
            <h1>Add new Article</h1>
            <FormInput label={"Category:"} value={category} onChangeValue={setCategory} />
            <FormInput label={"Title:"} value={title} onChangeValue={setTitle} />
            <FormInput label={"Author:"} value={author} onChangeValue={setAuthor} />
            <FormInput label={"Text:"} value={text} onChangeValue={setText} />
        </form>
    );
}