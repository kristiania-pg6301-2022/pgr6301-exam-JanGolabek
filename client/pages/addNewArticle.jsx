import React, { useContext,useState } from "react";
import { useNavigate } from "react-router-dom";
import { FormInput } from "../lib/formInput";
import {ArticlesApiContext} from "../articlesApiContext";

export function AddNewArticle() {
    const { createArticle } = useContext(ArticlesApiContext);
    const [category, setCategory] = useState("");
    const [title, setTitle] = useState("");
    const [author, setAuthor] = useState("");
    const [text, setText] = useState("");

    const navigate = useNavigate();

    function handleSubmit(event) {
        event.preventDefault();
        createArticle({ category, title, author, text });
        navigate("/");
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1>Add new Article</h1>

            <FormInput label={"Category:"} value={category} onChangeValue={setCategory} />
            <FormInput label={"Title:"} value={title} onChangeValue={setTitle} />
            <FormInput label={"Author:"} value={author} onChangeValue={setAuthor} />
            <FormInput label={"Text:"} value={text} onChangeValue={setText} />

            <div>
                <button>Submit</button>
            </div>
        </form>
    );
}