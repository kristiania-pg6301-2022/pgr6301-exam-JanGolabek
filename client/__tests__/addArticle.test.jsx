import { AddNewArticle } from "../addNewArticle";
import React from "react";
import ReactDOM from "react-dom";

describe("add article component", () => {
    it("shows article form", () => {
        const element = document.createElement("div");
        ReactDOM.render(<AddNewArticle />, element);
        expect(element.innerHTML).toMatchSnapshot();
        expect(
            Array.from(element.querySelectorAll("form label strong")).map(
                (e) => e.innerHTML
            )
        ).toEqual(["Category:", "Title:", "Author:", "Text:"]);
    });
});