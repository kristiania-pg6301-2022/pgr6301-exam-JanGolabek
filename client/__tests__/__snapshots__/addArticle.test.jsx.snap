// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add article component shows article form 1`] = `"<form><h1>Add new Article</h1><div><label><strong>Category:</strong> <input value=\\"\\"></label></div><div><label><strong>Title:</strong> <input value=\\"\\"></label></div><div><label><strong>Author:</strong> <input value=\\"\\"></label></div><div><label><strong>Text:</strong> <input value=\\"\\"></label></div></form>"`;
